pkg/module/service.go

```diff
@@ -11,6 +11,7 @@ import (
 type ServiceInterface interface {
 	Sync(ctx context.Context, r *domain.Resource) (*domain.Resource, error)
 	Validate(ctx context.Context, res *domain.Resource) error
+	Act(ctx context.Context, r *domain.Resource, action string, params map[string]interface{}) (map[string]interface{}, error)
```

Comment:
What about naming it as `Do` ?
Like the one that `net/http` [Client](https://pkg.go.dev/net/http#Client.Do) has

Reply:
Wanted to keep it to a form of the word `Action`. This seemed the closes verb to it.